### PR TITLE
dma: expand dma api for paging purpose.

### PR DESCRIPTION
**Pull request by inteljiangwe1**
- expand dma api for paging purpose
    - add host/imr as memory type
    - add polling mode in dma config

Signed-off-by: jiangwe1 <wei.w.jiang@intel.com>

### CODE REVIEW

File: include/drivers/dma.h [hunk 1]
```diff
@@ -32,7 +32,12 @@ enum dma_channel_direction {
 	MEMORY_TO_MEMORY = 0x0,
 	MEMORY_TO_PERIPHERAL,
 	PERIPHERAL_TO_MEMORY,
-	PERIPHERAL_TO_PERIPHERAL /*only supported in NXP EDMA*/
+	PERIPHERAL_TO_PERIPHERAL, /*only supported in NXP EDMA*/
+	/* intel dma dedicated features */
+	MEMORY_TO_HOSTDDR,
+	HOSTDDR_TO_MEMORY,
+	MEMORY_TO_IMR,
```
Comment (galak) [inline include/drivers/dma.h:39]: what does IMR stand for?

**Comment by galak** [comment]
Can you explain what is meant by "paging purpose" and what the use case is here?

File: include/drivers/dma.h [hunk 2]
```diff
@@ -142,7 +147,11 @@ typedef void (*dma_callback_t)(const struct device *dev, void *user_data,
  *     linked_channel       [ 20 : 26 ] - after channel count exhaust will
  *                                        initiate a channel service request
  *                                        at this channel
- *     reserved             [ 27 : 31 ]
+ *     polling_mode         [ 27 ]      - whether dma uses async or polling mode
```
Comment (galak) [inline include/drivers/dma.h:150]: Do we have some code or something here around `polling_mode`?

**Review by galak — CHANGES_REQUESTED** [review]
Need a bit more info before approving this.

File: include/drivers/dma.h (hunk 1 above)
Comment (nashif) [reply include/drivers/dma.h:39]: Isolated Memory Regions

**Comment by nashif** [comment]
@dcpleung @inteljiangwe1 can you provide some context here on how this relates to paging and why this is needed?

**Comment by inteljiangwe1** [comment]
In some intel IPs, there are limited SRAM and large IMR(or DDR) region, so store the outdated SRAM pages data in IMR, and exchange them back to SRAM with **DMA** transcation when demanded/visited. we call it demand-paging. It is triggered by page-fault, and excuted in its handler.
This patch grants the DMA capability copying data between SRAM and IMR/DDR, and could be run in page-fault handler

File: include/drivers/dma.h (hunk 2 above)
Comment (inteljiangwe1) [reply include/drivers/dma.h:150]: it is in our local project currently, may get them upstreamed in future

**Comment by inteljiangwe1** [comment]
@nashif @galak

**Pull request closed without merging**